test/integration/test_BAOABIntegrator/test_BAOABIntegrator.cpp

```diff
@@ -72,7 +73,8 @@ bool inject_test_potential(std::unique_ptr<mjolnir::SimulatorBase>& sim_base)
         return false;
     }
 
-    sim->forcefields().external().emplace(mjolnir::make_unique<interaction_type>(
+    forcefield_type* ff = dynamic_cast<forcefield_type*>(sim->forcefields().get());
+    ff->external().emplace(mjolnir::make_unique<interaction_type>(
```

Comment:
I think it is good to check the `dynamic_cast` succeeded.
```suggestion
    forcefield_type* ff = dynamic_cast<forcefield_type*>(sim->forcefields().get());
    assert(ff);
    ff->external().emplace(mjolnir::make_unique<interaction_type>(
```